dallinger/experiment_server/experiment_server.py

```diff
@@ -1260,15 +1263,21 @@ def transformation_post(node_id, info_in_id, info_out_id):
 
 
 @app.route("/notifications", methods=["POST", "GET"])
+@crossdomain(origin='*')
 def api_notifications():
     """Receive MTurk REST notifications."""
     event_type = request.values['Event.1.EventType']
-    assignment_id = request.values['Event.1.AssignmentId']
+    assignment_id = request.values.get('Event.1.AssignmentId')
+    participant_id = request.values.get('participant_id')
+    details = request.values.get('details')
+    if details:
+        details = loads(details)
 
     # Add the notification to the queue.
     db.logger.debug('rq: Queueing %s with id: %s for worker_function',
                     event_type, assignment_id)
-    q.enqueue(worker_function, event_type, assignment_id, None)
+    q.enqueue(worker_function, event_type, assignment_id,
```

Comment:
I thought a bit more about this and I think we should consider a slightly different approach for how to transmit tracking events to the server. Using the worker queue is a good idea that we should keep. However, given that these events are now infos associated with nodes, they are better handled not by the `/notifications/ POST` route but by the `/info/ POST` route. The logic would still be the same, just happening in a different place: if the `info_type` is `TrackingEvent`, enqueue the event.

Reply:
The `/info/` route currently blocks, so we'd still need to add a code path to enqueue the worker function there, which would add complexity to that route and duplicate code that's already in the `/notifications/` route. It's definitely more semantically obvious to do it via `/info/`, but it also means duplicating code and adding complexity.

Reply:
I think in this case it's worth duplicating the code (i.e., having both the notification route and the info route use the queue when appropriate). It adds complexity by adding a special case to the info route, but decreases complexity by making the notification route single-purpose: handling notifications.